crates/net/network/src/manager.rs

```diff
@@ -176,7 +178,11 @@ where
             hello_message,
             fork_filter,
         );
-        let state = NetworkState::new(client, discovery, peers_manger, genesis_hash);
+        let mut state = NetworkState::new(client, discovery, peers_manger, genesis_hash);
+
+        for NodeRecord { address, tcp_port, udp_port: _, id } in trusted_nodes {
+            state.add_peer_address(id, SocketAddr::from((address, tcp_port)));
+        }
```

Comment:
this  will create a bunch of useless events which we don't want.

now that I'm seeing this, since trusted nodes are basically only a `PeersManager` thing, perhaps we just move it to the `PeersConfig` or add to the new function of PeersManager.

the former feels cleaner, but is perhaps a bit less convenient in the `NetworkConfig`. However we'll revamp this eventually, so can just move into `PeersConfig` and we'll figure out NetworkConfig ergonomics later.